### PR TITLE
mvcc/txn: use structured log format and hex key format

### PR DESCRIPTION
**Pull request by huachaohuang**
## What have you changed? (mandatory)

Update the log format and the key format according to the RFCs.

Check the document for how the format work: https://docs.rs/slog/2.0.0-3.1/slog/macro.log.html#values-implementing-fmtdisplay

## What are the type of the changes? (mandatory)

The currently defined types are listed below, please pick one of the types for this PR by removing the others:
- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

Unit test

## Does this PR affect documentation (docs) update? (mandatory)

No

## Does this PR affect tidb-ansible update? (mandatory)

No

## Refer to a related PR or issue link (optional)

https://github.com/tikv/rfcs/blob/master/text/2018-12-19-unified-log-format.md
https://github.com/tikv/rfcs/blob/master/text/2018-11-12-unified-key-format.md

### CODE REVIEW

File: src/storage/mvcc/txn.rs [hunk 1]
```diff
@@ -215,8 +216,10 @@ impl<S: Snapshot> MvccTxn<S> {
                         // None: related Rollback has been collapsed.
                         // Rollback: rollback by concurrent transaction.
                         info!(
-                            "txn conflict (lock not found), key:{}, start_ts:{}, commit_ts:{}",
-                            key, self.start_ts, commit_ts
+                            "txn conflict (lock not found)";
+                            "key" => hex::encode_upper(key.as_encoded()),
```
Comment (zhangjinpeng1987) [inline src/storage/mvcc/txn.rs]: Can we use escape instead of hex?

**Comment by siddontang** [comment]
@huachaohuang 

> An alternative is to implement the slog::Value trait for types::Key so that we don't need to write hex::encode_upper(key.as_encoded()) everywhere, but it requires an extra key copy.

I like this way, writing `hex::encode_upper(key.as_encoded())` every where is verbose.

Btw, why do we need another key copy? If it can't be avoided, I still think it is acceptable because of the log frequency, it can't hurt the performance mostly.

File: src/storage/mvcc/txn.rs (hunk 1 above)
Comment (huachaohuang) [reply src/storage/mvcc/txn.rs]: We need to follow the "Unified Key Format" RFC to use the hex format.

**Comment by huachaohuang** [comment]
Oh, looks like I can implement `slog::Value` for `storage::types::Key` without an extra copy.

**Comment by huachaohuang** [comment]
@breeswish Already did.

**Comment by breeswish** [comment]
Looks nice!

**Comment by huachaohuang** [comment]
@breeswish @siddontang I think this is the optimal implementation for now, PTAL.

File: src/storage/mvcc/txn.rs (hunk 1 above)
Comment (siddontang) [reply src/storage/mvcc/txn.rs]: @zhangjinpeng1987 

Please remember to change other places which use escape.

**Comment by Hijiao** [comment]
Hi, I reduced one extra key copy in this PR. #4087

**Comment by huachaohuang** [comment]
Oh, perfect, you'are amazing! @Hijiao

**Pull request merged**